public/static/docs/user-guide/setup-google-drive-remote.md

```diff
@@ -0,0 +1,90 @@
+# Setup a Google Drive DVC Remote
+
+Follow this guide to setup Google Drive as your <abbr>DVC project</abbr>'s
+[remote storage](/doc/command-reference/remote).
+
+## Create and Configure a **Google Project**
+
+Since Google Drive has tight API usage quotas, one-time creation and
+configuration of a _Google Project_ is required. The mandatory
+[OAuth](https://developers.google.com/identity/protocols/OAuth2) credentials,
+**client ID** and **client secret**, can be generated only once. Please share
+them with other DVC users in your team to grant them access to Google's API.
+
+1. Log into the [Google Cloud Platform](https://console.developers.google.com)
+   account.
+
+2. Create a **New Project** or select an available one.
+
+3. From the left sidebar menu select **APIs & Services** -> **Dashboard** and
+   click on **ENABLE APIS AND SERVICES** to type and search for the **Google
+   Drive API**, in order to enable it from the search results.
+
+   ![](/static/img/gdrive-enable-apis-and-services.png)
+
+4. From the left sidebar menu select **APIs & Services** -> **OAuth consent
+   screen**, select a desired **User Type**, and click on the **Create** button.
+   On the next screen enter an **Application name** and click the **Save**
+   button. (You will be able to come back to edit others settings later.)
+
+5. From the left sidebar menu select **APIs & Services** -> **Credentials**, and
+   click the **Create credentials** button. Select the **OAuth client ID** ->
+   **Other** and click **Create** to proceed with a default suggested **Name**
+   for your new **OAuth client ID**.
+
+   ![](/static/img/gdrive-create-credentials.png)
+
+6. The newly generated **client ID** and **client secret** should be shown to
+   you. Please use these values below:
+
+  ```dvc
+  $ dvc remote add -d myremote gdrive://root/path/to/dir
+  $ dvc remote modify myremote gdrive_client_id my_gdrive_client_id
+  $ dvc remote modify myremote gdrive_client_secret gdrive_client_secret
+  ```
+
+  > Please see the [URL format](#url-format-for-google-drive-remotes) section
+  > below to learn how to form `gdrive://...` remote URLs.
+
+⚠️ It is safe to share **client ID** and **client secret** at with your team.
+These credentials are used to
+[retrieve an authorization token](https://developers.google.com/identity/protocols/OAuth2)
+and
+[authenticate your application](https://developers.google.com/drive/api/v2/about-auth)
+with Google. Credentials do not provide access to all of your Google Drive
+storage. Note that Google rate-limits the global number of queries per each
```

Comment:
Last Q on this first version (which I've already rewritten but haven't pushed just yet):

> Credentials do not provide access to all of your Google Drive
storage.

??? Cc @MaxRis

Reply:
i.e. why is it safe to share the OAuth credentials?

Reply:
@jorgeorpinel as it was discussed above there are 2 different sets of credentials: OAuth client id and secret credentials and user's GDrive access token credentials generated by DVC after user did login.
When we asks users to create OAuth credentials it is important to let them know that OAuth credentials can be created only once and provided to others team members. Also since the questions already raised ( it is suspicious for users is it safe to share with others so called client **secret**  ), we want explicitly mention that OAuth credentials doesn't provide access to user's GDrive data files and other sensitive information.

Reply:
I get that we want to clarify this common question about sharing credentials and if you review the new version I do have these notes both on the client ID/client secret and on the access token. What I don't understand (and should probably just research/try myself) is: "OAuth credentials doesn't provide access to user's GDrive data files and other sensitive information". Isn't that exactly what the OAuth credentials are for? To provide access to the GDrive folders and files! What data and sensitive info are you referring to @MaxRis? Thanks

p.s. "OAuth credentials can be created only once" I don't think this is true, at least effectively: you can just create more OAuth clients in the same GC project and delete the previous ones. I removed that note as I don't see how it's helpful for the users to know this but please lmk if I'm missing the point.

Reply:
Sorry for misunderstanding!
Well, initially, we have placed following explanation of **client id** and **client secret** into this doc:

```
⚠️ It is safe to share a **client id** and a **client secret** at least with
your teammates or inside an organization. These credentialds are used to
[retrieve an authorization token](https://developers.google.com/identity/protocols/OAuth2)
and
[authenticate your application](https://developers.google.com/drive/api/v2/about-auth)
to Google. Credentialds don't give an access to your Google Drive storage. But
Google limits global number of queires per each **client id** and uncontrolled
usage of your **client id** by outer world can lead to
[reaching the allowed Google API usage limits](https://developers.google.com/drive/api/v2/handle-errors?hl=ro#quota).
```

Above already explains how **client id** and **client secret** are used. Please let me know if there are still some questions about OAuth credentials.


> What data and sensitive info are you referring to @MaxRis? 

There I meant just any private data which is stored at user's GDrive and normally shouldn't be shared with anyone without user's allowance (to be specific, in this context, these are folders and files).

> p.s. "OAuth credentials can be created only once" I don't think this is true

You're right, sorry, I supposed to explain that if someone inside team or organization already created OAuth credentials it will be better share them with others users in team/organization to avoid difficulties for each new user to generate separate OAuth credentials.

I'm going to check the latest version of the doc, will let you know if anythings, thanks!

Reply:
I think the note may be misleading because whether sharing those creds it's safe or not may depends on how you configure the GC project or OAuth client: What APIs you enable, what folders it has access to, etc. Though maybe it's not a very important remark.

I understand that the creds are only used to get an access token, and that for that you need access to the Google Drive resources with your own user. I guess I just need to try expressing this more clearly, no worries I'll work on it.

And yes, please feel free share any feedback you have on the latest version 🙂  Keep in mind the URL part may still need another significant review but I haven't been able to try that part because of a bug in `dvc push` 🙁

Reply:
OK, reworded notes about sharing creds and token in 95de2824!